script/ide/System/Compiler/ast.lua

```diff
@@ -0,0 +1,113 @@
+----------------------------------------------------------------------
+-- Ast Class
+-- Author: Zhiyuan
+-- Date: 2016-12-22
+-- provide helper function to manipulate ast
+----------------------------------------------------------------------
+NPL.load("(gl)script/ide/System/Compiler/nplgen.lua");
+NPL.load("(gl)script/ide/System/Compiler/nplp.lua");
+
+local nplgen = commonlib.gettable("System.Compiler.nplgen")
+local ast = commonlib.inherit(nil, commonlib.gettable("System.Compiler.ast"))
+
+ast.content = {}
+ast.params= {}
+
+function ast:new(params, mode, con)
```

Comment:
if you use inherit to create a class table, you do not need to overwrite the new method. 
instead ctor()    if you want to init something  use :init()